### PR TITLE
make: orderedGrammar is .PRECIOUS

### LINKED ISSUE #14411: Failing "make doc_gram_rsts" deletes orderedGrammar
make/dune should not be deleting files here!  It's particularly unpleasant to lose your changes because changes to orderedGrammar are often exactly why you're making this target.

```
$ make doc_gram_rsts
     :
Warning: Nonterminal where not included in .rst files
TOTAL Nonterminals not included = 71
1 error(s)
Makefile.doc:305: recipe for target 'doc/tools/docgram/orderedGrammar' failed
make[1]: *** [doc/tools/docgram/orderedGrammar] Error 1
make[1]: *** Deleting file 'doc/tools/docgram/orderedGrammar'
```

Attn: @Zimmi48

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #14411

### CODE REVIEW

**Comment by jfehrle** [comment]
This works, thanks.  Maybe also make `doc/tools/docgram/updated_rsts` precious, although nothing depends on it (it's created as a flag).

File: Makefile.doc [hunk 1]
```diff
@@ -301,9 +301,12 @@ doc/tools/docgram/fullGrammar: $(DOC_GRAM) $(DOC_MLGS)
 	$(HIDE)$(DOC_GRAM) -short -no-warn $(DOC_MLGS)
 
 #todo: add a dependency of sphinx on updated_rsts when we're ready
-doc/tools/docgram/orderedGrammar doc/tools/docgram/updated_rsts: doc/tools/docgram/fullGrammar $(DOC_GRAM) $(DOC_EDIT_MLGS)
+doc/tools/docgram/orderedGrammar doc/tools/docgram/updated_rsts: #doc/tools/docgram/fullGrammar $(DOC_GRAM) $(DOC_EDIT_MLGS)
```
Comment (jfehrle) [inline Makefile.doc:304]: Keeping the removed items around as a comment?

File: Makefile.doc [hunk 2]
```diff
@@ -301,9 +301,12 @@ doc/tools/docgram/fullGrammar: $(DOC_GRAM) $(DOC_MLGS)
...
 	$(SHOW)'DOC_GRAM_RSTS'
 	$(HIDE)$(DOC_GRAM) $(DOCGRAMWARN) -check-cmds -check-tacs $(DOC_MLGS) $(DOC_RSTS)
+	false
```
Comment (jfehrle) [inline Makefile.doc:307]:
```suggestion
	$(HIDE)false
```

It looks odd to see "false" in the output.

**Comment by jfehrle** [comment]
OTOH, with this change, the only way to build `doc_grammar.exe` is AFAICT `make _build/default/doc/tools/docgram/doc_grammar.exe`.  That's not too good.

File: Makefile.doc (hunk 1 above)
Comment (SkySkimmer) [reply Makefile.doc]: this an the `false` were testing remains, now removed

**Comment by SkySkimmer** [comment]
>This works, thanks. Maybe also make doc/tools/docgram/updated_rsts precious, although nothing depends on it (it's created as a flag).

It doesn't sound precious to me.

**Comment by jfehrle** [comment]
I meant to say "depends on it yet".  The idea was to force rebuilding the rsts if the flag is set.  We'll reconsider that detail later.

**Review by Zimmi48 — APPROVED** [review]
Let's merge this now since it is already a useful fix as is.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Comment by Zimmi48** [comment]
@coqbot merge now

**Comment by jfehrle** [comment]
OK, but let's make it easier to rebuild `doc_grammar`.

**Comment by Zimmi48** [comment]
Yes. there is another PR open that does this.

**Pull request merged**